action.yml

```diff
@@ -12,13 +16,17 @@ inputs:
   jira-project-key:
     description: 'Jira project key where issues will be created'
     required: true
-  jira-base-uri:
-    description: 'URI of the target Jira API endpoint'
-    default: 'https://jira-base-uri'
-  jira-ignore-statuses:
+  jira-ignored-statuses:
     description: 'Comma-separated list of Jira issue statuses that will be ignored when syncing'
     required: false
-    default: 'Done'
+    # Default - "Done, Closed, Resolved"
+  jira-assignee:
+    description: 'the assignee accountId to assign the Jira ticket to'
+    required: false
+  jira-transition-map:
+    description: 'JSON string of Jira statuses and transition pairs to traverse when closing an issue e.g 'To Do:Start Progress, In Progress:Done'
```

Comment:
Just to clarify, the items in this map would "Status A: Status B, Status B: Status C", which implies "Status A transitions to Status B, and Status B transitions to Status C." So in your example is 'Start Progress' the same as 'In Progress'?

Reply:
Close, one subtle thing to note is that Status != Equal transition. Another is that the collection is unordered. Lets look at an example:

![image](https://github.com/Enterprise-CMCS/mac-fc-security-hub-visibility/assets/130405742/f78ce8ce-875e-4ea6-b230-94b15aa280c1)

In the example graph we have 2 terminal statuses, F and E, and lets say that the requirements are that given A the terminal transition should be F. The transition map would be:
'A:elk, B:ant, D:fox, C:dog'

Since the list isn't processed in any order, its just a lookup table 'C:dog, B:ant, D:fox, , A:elk' would be an equivalent map. If your status is A it doesn't matter if A is first or last in the list, you just find your own status and then execute the 'elk' transition.

So in the example its implied that the workflow is something simple like To Do -- [Start Progress] --> In Progress -- [Done] --> Completed.  With the transitions in brackets.

I thought this might be an issue. Its a bit too complicated to explain in a comment for a config file. If you didn't have any other issues with this approach, I was expecting that I'd have to update the README file with some diagrams and detailed examples.

Which I'm mixed on the approach. I didn't write the docs because I thought we might decide on making some big changes around transitions in the code review. So if you have an criticism/alternative ideas around this feel free.